inc/config.php

```diff
@@ -49,28 +49,76 @@
 
 Config::detectRootDoc();
 
-if (!file_exists(GLPI_CONFIG_DIR . "/config_db.php")) {
+$conf_exists = file_exists(GLPI_CONFIG_DIR . "/db.yaml");
+if (!$conf_exists && file_exists(GLPI_CONFIG_DIR . "/config_db.php")) {
+    //convert old config file to new one
+    $oldconf = file_get_contents(GLPI_CONFIG_DIR . "/config_db.php");
+    $matches = [];
+
+    preg_match('/dbhost\s*=\s*["\'](.+)["\'];/', $oldconf, $matches);
+    $host = $matches[1];
+
+    preg_match('/dbuser\s*=\s*["\'](.+)["\'];/', $oldconf, $matches);
+    $user = $matches[1];
+
+    preg_match('/dbpassword\s*=\s*["\'](.+)["\'];/', $oldconf, $matches);
+    $password = $matches[1];
+
+    preg_match('/dbdefault\s*=\s*["\'](.+)["\'];/', $oldconf, $matches);
+    $dbname = $matches[1];
+
+    $yaml = [
+    ];
+    $migrated = \DBConnection::createMainConfig(
+        'mysql',
+        $host,
+        $user,
+        $password,
+        $dbname
+    );
+    if (!$migrated) {
+        if (!isCommandLine()) {
```

Comment:
I think you can replace all this if/else code and replace it by `$conf_exists = false;` and maybe a message to indicate that migration fails.

Reply:
messages are not really the same, I'd prefer to let it as is for now